src/sbvr-api/sbvr-utils.ts

```diff
@@ -1489,183 +1468,174 @@ const runGet = (
-			});
+			};
 		} else {
 			// TODO: request.resourceName can be '$serviceroot' or a resource and we should return an odata xml document based on that
-			return Bluebird.resolve({
+			return {
 				status: 404,
-			});
+			};
 		}
 	}
 };
 
-const runPost = (
+const runPost = async (
 	_req: _express.Request,
 	_res: _express.Response,
 	request: uriParser.ODataRequest,
 	tx: _db.Tx,
-) => {
+): Promise<number | undefined> => {
 	const vocab = request.vocabulary;
 
 	const { idField } = getAbstractSqlModel(request).tables[
 		resolveSynonym(request)
 	];
 
-	return runQuery(tx, request, undefined, idField)
-		.tap(sqlResult => {
-			if (sqlResult.rowsAffected === 0) {
-				throw new PermissionError();
-			}
-			return validateModel(tx, vocab, request);
-		})
-		.then(({ insertId }) => insertId);
+	const sqlResult = await runQuery(tx, request, undefined, idField);
+	if (sqlResult.rowsAffected === 0) {
+		throw new PermissionError();
+	}
+	await validateModel(tx, vocab, request);
+
+	return sqlResult.insertId;
 };
 
-const respondPost = (
+const respondPost = async (
 	req: _express.Request,
 	res: _express.Response,
 	request: uriParser.ODataRequest,
 	id: number,
 	tx: _db.Tx,
-): Bluebird<Response> => {
+): Promise<Response> => {
 	const vocab = request.vocabulary;
 	const location = odataResponse.resourceURI(vocab, request.resourceName, id);
 	if (DEBUG) {
 		api[vocab].logger.log('Insert ID: ', request.resourceName, id);
 	}
-	return Bluebird.try(() => {
-		const onlyId = { d: [{ id }] };
-		if (
-			location == null ||
-			_.includes(
-				['0', 'false'],
-				_.get(request, ['odataQuery', 'options', 'returnResource']),
-			)
-		) {
-			return onlyId;
+
+	let result: AnyObject = { d: [{ id }] };
+	if (
+		location != null &&
+		!_.includes(
+			['0', 'false'],
+			_.get(request, ['odataQuery', 'options', 'returnResource']),
+		)
+	) {
+		try {
+			result = (await runURI('GET', location, undefined, tx, req)) as AnyObject;
+		} catch {
+			// If we failed to fetch the created resource then we use just the id as default.
 		}
-		return (
-			runURI('GET', location, undefined, tx, req)
-				// If we failed to fetch the created resource then just return the id.
-				.catchReturn(onlyId)
-		);
-	}).then((result: AnyObject) => {
-		return runHooks('PRERESPOND', request.hooks, {
-			req,
-			res,
-			request,
-			result,
-			tx,
-		}).return({
-			status: 201,
-			body: result.d[0],
-			headers: {
-				contentType: 'application/json',
-				Location: location,
-			},
-		});
+	}
+
+	await runHooks('PRERESPOND', request.hooks, {
+		req,
+		res,
+		request,
+		result,
+		tx,
 	});
+
+	return {
+		status: 201,
+		body: result.d[0],
+		headers: {
+			contentType: 'application/json',
+			Location: location,
+		},
+	};
 };
 
-const runPut = (
+const runPut = async (
 	_req: _express.Request,
 	_res: _express.Response,
 	request: uriParser.ODataRequest,
 	tx: _db.Tx,
-): Bluebird<undefined> => {
+): Promise<undefined> => {
```

Comment:
```suggestion
): Promise<void> => {
```

?

Reply:
The `undefined` is specifically used elsewhere, basically these `runX` functions can return `_db.Result | number | undefined`